dpytools/slack/slack.py

```diff
@@ -1,14 +1,25 @@
+import os
+import logging
+from dpytools.http_clients.base import BaseHttpClient
 
 class SlackNotifier:
 
     def __init__(self):
-        # Set a webhok via an env var, ask Mike for a 
-        #web hook url.
-        ...
-
-    def notify(self, msg: str):
-        # Check formatting options for messages to slack.
-        # From memory you style it via sending a dictionary.
-        # It's a post request so do use the http client
-        # we've developing elsewhere in this library.
-        ...
\ No newline at end of file
+        self.webhook_url = os.getenv("SLACK_WEBHOOK_URL")
+        if not self.webhook_url:
+            raise ValueError('SLACK_WEBHOOK_URL is not set')
+        self.http_client = BaseHttpClient()
+        self.validate_webhook_url()
+
+    def validate_webhook_url(self):
+        response = self.http_client.get(self.webhook_url)
+        if response.status_code != 200:
+            logging.error(f'Invalid SLACK_WEBHOOK_URL: {response.status_code}')
+            raise ValueError('Invalid SLACK_WEBHOOK_URL')
+
+    def notify(self, msg: dict):
```

Comment:
can we rename this to `msg_dict` and include some comments to include a link to where the dict structure is documented please.

can we **also** add a new method of `msg_str` which just takes a string and puts it into the dict in the right place (i.e wrap it so people can quickly punt a string message at the webhook).